### PR TITLE
Fix for the huge audio latency (>200 ms) for the …

### LINKED ISSUE #6965: SamplePlayer latency - it's just bad performance.
**Operating system or device - Godot version:**
Godot 2.1-stable on Xubuntu 14.10 64-bit. Project also tested on Android 4 to 6.

**Issue description** (what happened, and what was expected):
Delay between action on the screen (for example clicking the button) and hearing the sounds, that is played by pressing the button with something as simple as

```
func _on_TouchScreenButton_pressed():
    get_node("SamplePlayer").play("sample")
```

 is terribly long.
From my experiments it's around 285 ms.
godot_audio_problem_3

Latency like that basically prevents people (who don't know how to use low level audio AudioServer), from creating interactive audio games.

### PR DESCRIPTION
**Pull request by lonesurv1vor**
…PulseAudio, ALSA and RtAudio driver.

Partly fixes issue #6965. I made a post about this at there.

EDIT: I also cleaned up the code (mostly whitespace related stuff) for better readability (that is why there are so many changes in the files). And PulseAudio and ALSA do now actually use the global setting "audio/mix_rate" for the sampling rate instead of a fixed value of 44100.

(*bugsquad edit* - make it so that the issue doesn't get autoclosed if this is merged)

### CODE REVIEW

File: drivers/pulseaudio/audio_driver_pulseaudio.cpp [hunk 1]
```diff
@@ -60,7 +71,7 @@ Error AudioDriverPulseAudio::init() {
                           "Sound",      // stream description
                           &spec,
                           NULL,         // use default channel map
-                          NULL,         // use default buffering attributes
+                          &attr,         // use default buffering attributes
```
Comment (akien-mga) [inline drivers/pulseaudio/audio_driver_pulseaudio.cpp]: The comment is no longer true no?

Comment (lonesurvivor) [inline drivers/pulseaudio/audio_driver_pulseaudio.cpp]: Yeah, forgot about that. Changed.

**Comment by lonesurvivor** [comment]
Fixed the ALSA driver as well.

**Comment by lonesurvivor** [comment]
Now with fixed RtAudio as well. Look at the issue #6965 for explanation. Please test under Windows and Mac if you can.

**Comment by akien-mga** [comment]
Awesome work! :)

**Pull request merged**